unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+import uuid
+from unittest import mock
+
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+from unittest.mock import patch
+
+area_id = str(uuid.uuid4())
+area_uid = None
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.publish = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe = MagicMock()
+
+
+@pytest.fixture
+def redis_client_auto_register():
+    return RedisClientBase(area_id=area_id, autoregister=False)
+
+
+class TestRedisClientBase:
+    @classmethod
+    def setup_class(cls):
+        cls.obj = RedisClientBase(area_id=area_id, autoregister=False)
+
+    def test_constructor_registry_success(self, pubsub_thread=None):
+        assert self.obj.area_id == area_id
+        assert self.obj.area_uuid == area_uid
+        assert self.obj.is_active == False
+        assert self.obj._subscribed_aggregator_response_cb is None
+        assert self.obj._subscribe_to_response_channels(pubsub_thread) is None
```

Comment:
That doesn't seems like a useful assertion. What could be actually be checked here if thread is increased by one after calling this method.

You can call [threading.active_count()](https://docs.python.org/3/library/threading.html#threading.active_count) before and after this line and assert that number of active threads have increase exactly by 1

Reply:
I agree with Faizan. Also, it is in general recommended not to test private methods/variables directly. We mostly care about the output and the side-effects of our public methods.
Suppose we unit-test a public method and we assert its output. We don't care about how many private methods are called inside this public method, as long as the output is what we expect. This saves us a lot of time and it makes the code more flexible.

I highly encourage you to have a look at this amazing talk by Sandi Metz: https://www.youtube.com/watch?v=URSWYvyc42M&t=947s